packages/cli/src/NodeTypes.ts

```diff
@@ -25,7 +26,15 @@ class NodeTypesClass implements INodeTypes {
 	}
 
 	getAll(): INodeType[] {
-		return Object.values(this.nodeTypes).map((data) => data.type);
+		return Object.values(this.nodeTypes).map((data) => {
+			try {
+				data.type.description.codex = this.getCodexSearchProperties(data.sourcePath);
```

Comment:
getAll is called with every api request
https://github.com/n8n-io/n8n/blob/master/packages/cli/src/Server.ts#L745

we should instead do this only once when initializing node types and not have to read from file with every request

maybe here https://github.com/n8n-io/n8n/blob/master/packages/cli/src/NodeTypes.ts#L14

though this NodeTypes is initialized for multiple commands
https://github.com/n8n-io/n8n/search?q=await+NodeTypes+init

we only really need this codex info for the start process, not other processes. so maybe we should only read codex info for our start process to avoid impacting performance of other processes

so maybe the best place to add codex info behind a boolean is here 
https://github.com/n8n-io/n8n/blob/master/packages/cli/src/LoadNodesAndCredentials.ts#L226
where we load data for nodes

and here to support custom packages 
https://github.com/n8n-io/n8n/blob/master/packages/cli/src/LoadNodesAndCredentials.ts#L198

might be best to check with someone who's more familiar with cli and how it works than me